tests/framework/base/ComponentTest.php

```
@@ -10,405 +10,880 @@
-        $component->{'as f'} = function () {
-            return new NewBehavior();
-        };
-        $this->assertNotNull($component->getBehavior('f'));
+
+        $component->{'as f'} = static fn() => new NewBehavior();
+
+        self::assertNotNull(
+            $component->getBehavior('f'),
+            'Should be attached via closure factory.',
+        );
     }
```

PHPMD (2.15.0)

[warning] The method testAttachBehavior() has 124 lines of code. Current threshold is set to 100. Avoid really long methods. (undefined)

(ExcessiveMethodLength)